### PR TITLE
Community: LlamaCppEmbeddings `embed_documents` and `embed_query`

### PR DESCRIPTION
**Pull request by keenborder786**
- **Description:** `embed_documents` and `embed_query` was throwing off the error as stated in the issue. The issue was that `Llama` client is returning the embeddings in a nested list which is not being accounted for in the current implementation and therefore the stated error is being raised.
- **Issue:** #28813

### CODE REVIEW

**Review by ccurme — COMMENTED** [review]
Could you add a test?

**Comment by keenborder786** [comment]
@ccurme OKAY

**Comment by keenborder786** [comment]
@ccurme please review

File: libs/community/langchain_community/embeddings/llamacpp.py [hunk 1]
```diff
@@ -116,7 +117,14 @@ def embed_documents(self, texts: List[str]) -> List[List[float]]:
             List of embeddings, one for each text.
         """
         embeddings = self.client.create_embedding(texts)
-        return [list(map(float, e["embedding"])) for e in embeddings["data"]]
+        if not isinstance(embeddings["data"][0]["embedding"][0], list):
```
Comment (ccurme) [inline libs/community/langchain_community/embeddings/llamacpp.py:120]: is it possible for this line to raise IndexError where it previously would not? e.g., if `texts` is empty or contains empty strings.

**Review by ccurme — COMMENTED** [review]
Thank you!

File: libs/community/langchain_community/embeddings/llamacpp.py (hunk 1 above)
Comment (keenborder786) [reply libs/community/langchain_community/embeddings/llamacpp.py:120]: ok

**Comment by keenborder786** [comment]
@ccurme please review

File: libs/community/langchain_community/embeddings/llamacpp.py (hunk 1 above)
Comment (ccurme) [reply libs/community/langchain_community/embeddings/llamacpp.py]: Sorry for the confusion. I didn't mean to *request* that we raise an error where it previously would not.

From an ignorant reading of the code (i.e., no knowledge of the behavior of the client) the update introduced the potential to raise IndexError where it previously wouldn't. I've pushed a change that passes your tests. Let me know if you see any issues with it.

**Review by ccurme — APPROVED** [review]

**Pull request merged**